pkg/endpoint/bpf.go

```diff
@@ -1327,6 +1329,38 @@ func (e *Endpoint) endpointPolicyLockdown() error {
 	return nil
 }
 
+// DisableDNSProxy tells the endpoint to force-remove any BPF policymap entries
+// that would proxy traffic to the DNS proxy. They are replaced with allow entries.
+//
+// This is intended to be called on shutdown, so that DNS traffic may still be allowed
+// while the agent is being upgaded.
+func (e *Endpoint) DisableDNSProxy() {
+	if e.IsProxyDisabled() {
+		return
+	}
+	if err := e.lockAlive(); err != nil {
```

Comment:
Is it possible that an endpoint would not be "alive" any more when this is called?